src/main/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingService.kt

```diff
@@ -60,20 +58,20 @@ class PrintResponseProcessingService(
 
     private fun processBatchCertificates(batchResponse: BatchResponse, certificates: List<Certificate>) {
```

Comment:
@Neil-Massey, I think this method will work properly for multiple `PrintRequests`, but the way we query certificates and pass into this method needs to change.

Here is the code that calls this method:

<pre>
    @Transactional
    fun processBatchResponses(batchResponses: List<BatchResponse>) {
        batchResponses.forEach { batchResponse ->
            val certificates =
               <b> certificateRepository.findByStatusAndPrintRequestsBatchId(
                    SENT_TO_PRINT_PROVIDER,
                    batchResponse.batchId
                )</b>
            processBatchCertificates(batchResponse, certificates)
            certificateRepository.saveAll(certificates)
        }
    }
</pre>

As you can see this queries certificates by their status not querying certificates containing print requests with status `SENT_TO_PRINT_PROVIDER`. So even if there is a print request with the status `SENT_TO_PRINT_PROVIDER` for a given certificate the status of the certificate itself might not reflect it, because that print request status might not be the last status for that certificate. So we lose the updates on that print request with the current implementation.

I think we should query based on print request status above, instead of certificate status to get the certificates to be updated.

Reply:
Agree this one could run into issues processing multiple print requests in parallel.  There is no status on the PrintRequest - am more inclined to remove the status from the initial query.

Reply:
Refactored to return certificates based on the Print Request status and batch.